include/picongpu/particles/manipulators/unary/FreeTotalCellOffset.hpp

```diff
@@ -0,0 +1,160 @@
+/* Copyright 2017-2018 Rene Widera, Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/particles/manipulators/unary/FreeTotalCellOffset.def"
+#include "picongpu/particles/functor/misc/TotalCellOffset.hpp"
+
+#include <string>
+
+
+namespace picongpu
+{
+namespace particles
+{
+namespace manipulators
+{
+namespace unary
+{
+namespace acc
+{
+    template< typename T_Functor >
+    struct FreeTotalCellOffset : private T_Functor
+    {
+
+        using Functor = T_Functor;
+
+        HDINLINE FreeTotalCellOffset(
+            Functor const & functor,
+            DataSpace< simDim > const & superCellToLocalOriginCellOffset
+        ) :
+            T_Functor( functor ),
+            m_superCellToLocalOriginCellOffset( superCellToLocalOriginCellOffset )
+        {
+        }
+
+        /** call user functor
+         *
+         * The random number generator is initialized with the first call.
+         *
+         * @tparam T_Particle type of the particle to manipulate
+         * @tparam T_Args type of the arguments passed to the user functor
+         * @tparam T_Acc alpaka accelerator type
+         *
+         * @param alpaka accelerator
+         * @param particle particle which is given to the user functor
```

Comment:
REQUIRED:
remove double particle

Reply:
no, this is doxygen.
the first is the lookup, the second the description

Reply:
Okay - sorry :+1: